### PR TITLE
validate DNS authority using dnspython

### PR DESCRIPTION
**Pull request by danielerez**
Used dns.resolver (dnspython) to validate DNS authority and resolvability.

Note: added 'dnspython' package to requirements, so a new image of test-infra should be pushed to quay.

### CODE REVIEW

**Comment by danielerez** [comment]
@romfreiman @tsorya

File: discovery-infra/start_discovery.py [hunk 1]
```diff
@@ -169,6 +170,31 @@ def _create_node_details(cluster_name):
...
 
 
+def validate_dns(client, cluster_id):
+    if args.managed_dns_domains == ":":
+        # 'set_dns' (using dnsmasq) is invoked after nodes_flow
+        return
+
+    cluster = client.cluster_get(cluster_id)
+    api_address = "api.{}.{}".format(cluster.name, cluster.base_dns_domain)
+    ingress_address = "ingress.apps.{}.{}".format(cluster.name, cluster.base_dns_domain)
```
Comment (tsorya) [inline discovery-infra/start_discovery.py:180]: lets add log with dns domain name.
Something like , validating domains ... exists.

File: requirements.txt [hunk 2]
```diff
@@ -8,3 +8,4 @@ requests==2.20.0
 retry==0.9.2
 tqdm==4.46.0
 waiting==1.4.1
+dnspython==1.16.0
```
Comment (tsorya) [inline requirements.txt:11]: we will have lint validation and it verifies that packages in requirements are in alphabetical order

File: discovery-infra/start_discovery.py (hunk 1 above)
Comment (danielerez) [reply discovery-infra/start_discovery.py:180]: done

File: requirements.txt (hunk 2 above)
Comment (danielerez) [reply requirements.txt]: done

**Pull request merged**